accesstoken2/src/lib/config-service.ts

```diff
@@ -0,0 +1,66 @@
+import { SSMClient, GetParameterCommand } from "@aws-sdk/client-ssm";
+
+const SESSION_TABLE_NAME_KEY = "SessionTableName";
+const DEFAULT_AUTHORIZATION_CODE_TTL_IN_SECS = 600;
+const PARAMETER_PREFIX = process.env.AWS_STACK_NAME || "";
+
+export class ConfigService {
+    private readonly authorizationCodeTtlInMillis: number;
+    private readonly configEntries: Map<string, string>;
+
+    constructor(private ssmClient: SSMClient) {
+        const envAuthCodeTtl = parseInt(process.env.AUTHORIZATION_CODE_TTL || "", 10);
+        this.authorizationCodeTtlInMillis =
+            (Number.isInteger(envAuthCodeTtl) ? envAuthCodeTtl : DEFAULT_AUTHORIZATION_CODE_TTL_IN_SECS) * 1000;
+        this.configEntries = new Map<string, string>();
+    }
+
+    public init(): Promise<string[]> {
+        const defaultKeys = [SESSION_TABLE_NAME_KEY];
+        const promises = [];
+        for (const key of defaultKeys) {
+            promises.push(this.getParameter(key));
+        }
+
+        return Promise.all(promises);
+    }
+
+    public async getRedirectUri(clientId: string) {
+        if (!clientId) {
+            throw new Error("Undefined clientId supplied");
+        }
+        return await this.getParameter(`clients/${clientId}/jwtAuthentication/redirectUri`);
+    }
+
+    public async getSessionTableName() {
+        return await this.getParameter(SESSION_TABLE_NAME_KEY);
```

Comment:
Few things here:

- this is returning the following in pseudo code
```
(Outside call)
          | Promise<getSessionTableName()>
                                              | Promise<getParameter()>
```

We don't NEED to have this function be async await. As we're going to be returning the promise created by `this.getParameter()`.

Which brings me to my second point:

- This function call isn't needed. 
We should be loading SessionTableNameKey on initiation as its something that's confirmed and not subject to change based on request.

instead, we should just access the configEntries map and return it from there. Which wont require a promise.

This code will actually be 'slow' as its a Promise and will be stuck at the end of an internal nodeJS queue to be executed. (it might not be noticeable but its still slower than it could be).

I would suggest changing this to be 

```
return this.configEntries.get(...);